pkg/controller/inferenceservice/resources/knative/service.go

```diff
@@ -144,6 +144,13 @@ func (c *ServiceBuilder) CreatePredictorService(name string, metadata metav1.Obj
 		addLoggerContainerPort(container)
 	}
 
+	predictor := ""
```

Comment:
nit: I'd call this endpoint.

You can also simplify with

endpoint := constants.InferenceServiceDefault
if isCanary {
  endpoint = constants.InferenceServiceCanary
}

Reply:
:+1: have changed this